app/models/event.rb

```diff
@@ -3,6 +3,10 @@
 
 # Records events in the lifecycle of a deposit
 class Event < ApplicationRecord
-  belongs_to :work
+  belongs_to :eventable, polymorphic: true
   belongs_to :user
+
+  validates :event_type, presence: true, event_type: true
```

Comment:
I don't understand the need for this.  Can you explain how an event could be set to something invalid?

Reply:
With the other changes made in this commit, I'm not sure it can other than through manual event creation (in the console), which is prolly not worth caring about. I'll remove the validation.